Request with too big header section is sent on big update

I am trying to run a migration, which updates a few properties on a lot of different nodes. If the session is not saved after every single node, I am seeing the following error in the jackrabbit log:

```
2015-08-12 08:56:30.861 ERROR [1447510073@qtp-1720435669-275] HttpMultipartPost.java:95 Error while processing multipart.
org.apache.commons.fileupload.FileUploadException: Header section has more than 10240 bytes (maybe it is not properly terminated)
	at org.apache.commons.fileupload.FileUploadBase.parseRequest(FileUploadBase.java:381) ~[jackrabbit-standalone-2.8.0.jar:na]
	at org.apache.commons.fileupload.servlet.ServletFileUpload.parseRequest(ServletFileUpload.java:126) ~[jackrabbit-standalone-2.8.0.jar:na]
	at org.apache.jackrabbit.server.util.HttpMultipartPost.extractMultipart(HttpMultipartPost.java:90) [jackrabbit-standalone-2.8.0.jar:na]
	at org.apache.jackrabbit.server.util.HttpMultipartPost.<init>(HttpMultipartPost.java:54) [jackrabbit-standalone-2.8.0.jar:na]
	at org.apache.jackrabbit.server.util.RequestData.<init>(RequestData.java:36) [jackrabbit-standalone-2.8.0.jar:na]
	at org.apache.jackrabbit.server.remoting.davex.JcrRemotingServlet.doPost(JcrRemotingServlet.java:387) [jackrabbit-standalone-2.8.0.jar:na]
	at org.apache.jackrabbit.webdav.server.AbstractWebdavServlet.execute(AbstractWebdavServlet.java:354) [jackrabbit-standalone-2.8.0.jar:na]
	at org.apache.jackrabbit.webdav.server.AbstractWebdavServlet.service(AbstractWebdavServlet.java:291) [jackrabbit-standalone-2.8.0.jar:na]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:820) [jackrabbit-standalone-2.8.0.jar:na]
	at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:511) [jackrabbit-standalone-2.8.0.jar:na]
	at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:390) [jackrabbit-standalone-2.8.0.jar:na]
	at org.mortbay.jetty.security.SecurityHandler.handle(SecurityHandler.java:216) [jackrabbit-standalone-2.8.0.jar:na]
	at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:182) [jackrabbit-standalone-2.8.0.jar:na]
	at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:765) [jackrabbit-standalone-2.8.0.jar:na]
	at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:418) [jackrabbit-standalone-2.8.0.jar:na]
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152) [jackrabbit-standalone-2.8.0.jar:na]
	at org.mortbay.jetty.handler.RequestLogHandler.handle(RequestLogHandler.java:49) [jackrabbit-standalone-2.8.0.jar:na]
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152) [jackrabbit-standalone-2.8.0.jar:na]
	at org.mortbay.jetty.Server.handle(Server.java:326) [jackrabbit-standalone-2.8.0.jar:na]
	at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:542) [jackrabbit-standalone-2.8.0.jar:na]
	at org.mortbay.jetty.HttpConnection$RequestHandler.content(HttpConnection.java:938) [jackrabbit-standalone-2.8.0.jar:na]
	at org.mortbay.jetty.HttpParser.parseNext(HttpParser.java:755) [jackrabbit-standalone-2.8.0.jar:na]
	at org.mortbay.jetty.HttpParser.parseAvailable(HttpParser.java:212) [jackrabbit-standalone-2.8.0.jar:na]
	at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:404) [jackrabbit-standalone-2.8.0.jar:na]
	at org.mortbay.jetty.bio.SocketConnector$Connection.run(SocketConnector.java:228) [jackrabbit-standalone-2.8.0.jar:na]
	at org.mortbay.thread.QueuedThreadPool$PoolThread.run(QueuedThreadPool.java:582) [jackrabbit-standalone-2.8.0.jar:na]
Caused by: org.apache.commons.fileupload.MultipartStream$MalformedStreamException: Header section has more than 10240 bytes (maybe it is not properly terminated)
	at org.apache.commons.fileupload.MultipartStream.readHeaders(MultipartStream.java:542) ~[jackrabbit-standalone-2.8.0.jar:na]
	at org.apache.commons.fileupload.FileUploadBase$FileItemIteratorImpl.findNextItem(FileUploadBase.java:976) ~[jackrabbit-standalone-2.8.0.jar:na]
	at org.apache.commons.fileupload.FileUploadBase$FileItemIteratorImpl.<init>(FileUploadBase.java:942) ~[jackrabbit-standalone-2.8.0.jar:na]
	at org.apache.commons.fileupload.FileUploadBase.getItemIterator(FileUploadBase.java:331) ~[jackrabbit-standalone-2.8.0.jar:na]
	at org.apache.commons.fileupload.FileUploadBase.parseRequest(FileUploadBase.java:349) ~[jackrabbit-standalone-2.8.0.jar:na]
	... 25 common frames omitted
```

I have checked the request that is causing this error, and its body looks something like this:

```
--c48f09d2ebd7f565002cf6ed304f6d98
--c48f09d2ebd7f565002cf6ed304f6d98
[repeated 500 times]
Content-Disposition: form-data; name=":diff"
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

+/jcr:versions/201507231648 : {"jcr:primaryType":"phpcrmig:version",}
^/cmf/candulor/contents/i18n:de_ch-excerpt-tags : []
^/cmf/candulor/contents/i18n:de_at-navContexts : []
[500 more lines]
--c48f09d2ebd7f565002cf6ed304f6d98--
```

The line with `--c48f09d2ebd7f565002cf6ed304f6d98` at the beginning is written once for every single change, which probably causes the error from above. As far as I can see this is caused by this line: https://github.com/jackalope/jackalope-jackrabbit/blob/3d3ba7cc920de810b8b7020bc4b35d5b9661d9ec/src/Jackalope/Transport/Jackrabbit/Client.php#L2052

Will try to find out what exactly is causing this, and to provide a test case.

--- a/src/Jackalope/Transport/Jackrabbit/Client.php
+++ b/src/Jackalope/Transport/Jackrabbit/Client.php
@@ -2088,7 +2088,6 @@ class Client extends BaseTransport implements QueryTransport, PermissionInterfac
         $data = '';
 
         $eol = "\r\n";
-        $data .= '--' . $mime_boundary . $eol ;
         if (is_array($value)) {
             if (is_array($value[0])) {
                 foreach ($value[0] as $v) {
@@ -2097,6 +2096,7 @@ class Client extends BaseTransport implements QueryTransport, PermissionInterfac
 
                 return $data;
             }
+            $data .= '--' . $mime_boundary . $eol ;
 
             if (is_resource(($value[0]))) {
                 $data .= 'Content-Disposition: form-data; name="' . $name . '"; filename="' . $name . '"' . $eol;
@@ -2125,6 +2125,7 @@ class Client extends BaseTransport implements QueryTransport, PermissionInterfac
 
                 return $data;
             }
+            $data .= '--' . $mime_boundary . $eol ;
             $data .= 'Content-Disposition: form-data; name="'.$name.'"'. $eol;
             $data .= 'Content-Type: text/plain; charset=UTF-8'. $eol;
             $data .= 'Content-Transfer-Encoding: 8bit'. $eol. $eol;

